webots_ros2_epuck/launch/robot_launch.py

```diff
@@ -17,35 +17,113 @@
 """Launch Webots e-puck driver."""
 
 import os
+import pathlib
+import launch
 from launch.substitutions import LaunchConfiguration
 from launch.actions import DeclareLaunchArgument
 from launch.substitutions.path_join_substitution import PathJoinSubstitution
-from launch.actions import IncludeLaunchDescription
-from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch_ros.actions import Node
 from launch import LaunchDescription
 from ament_index_python.packages import get_package_share_directory
+from webots_ros2_driver.webots_launcher import WebotsLauncher
 
 
 def generate_launch_description():
     package_dir = get_package_share_directory('webots_ros2_epuck')
     world = LaunchConfiguration('world')
+    robot_description = pathlib.Path(os.path.join(package_dir, 'resource', 'epuck_webots.urdf')).read_text()
+    ros2_control_params = os.path.join(package_dir, 'resource', 'ros2_control.yml')
+    use_sim_time = LaunchConfiguration('use_sim_time', default=True)
 
-    webots = IncludeLaunchDescription(
-        PythonLaunchDescriptionSource(
-            os.path.join(get_package_share_directory('webots_ros2_core'), 'launch', 'robot_launch.py')
-        ),
-        launch_arguments=[
-            ('package', 'webots_ros2_epuck'),
-            ('executable', 'driver'),
-            ('world', PathJoinSubstitution([package_dir, 'worlds', world])),
+    webots = WebotsLauncher(
+        world=PathJoinSubstitution([package_dir, 'worlds', world])
+    )
+
+    # TODO: Revert once the https://github.com/ros-controls/ros2_control/pull/444 PR gets into the release
+    sleep_time = 60 if 'CI' in os.environ else 10
+    controller_manager_timeout = ['--controller-manager-timeout', '50'] if os.name == 'nt' else []
+    controller_manager_prefix = 'python.exe' if os.name == 'nt' else f"bash -c 'sleep {sleep_time}; $0 $@' "
```

Comment:
https://github.com/ros-controls/ros2_control/pull/444 is released. We can use it as here:
https://github.com/cyberbotics/webots_ros2/blob/2d175978bfd3382dbce58b266532d6a111a36a24/webots_ros2_universal_robot/launch/robot_launch.py#L45-L46

Reply:
Ok, I have not noticed it, thanks!